### PR TITLE
FILTERS: FindTriangleGeom[Shapes|Sizes|Centroids] + SharedFeatureFace

### PR DESCRIPTION
**Pull request by imikejackson**
## Naming Conventions

Naming of variables should descriptive where needed. Loop Control Variables can use `i` if warranted. Most of these conventions are enforced through the clang-tidy and clang-format configuration files.

- [x] Class and Structs are `UpperCamelCase`
- [x] Class private member variables are `m_UpperCamelCase`
- [x] Class methods are `lowerCamelCase`
- [x] Method arguments are `lowerCamelCase`
- [x] Normal variables are `lowerCamelCase`
- [x] Constants are `k_UpperCamelCase`
- [x] Global statics are `s_UpperCamelCase`
- [x] Free Functions are `UpperCamelCase`
- [x] Macros are `ALL_UPPER_SNAKE_CASE`
- [x] Unit test will test data output from the filter.
- [x] Reused strings should be constants in an anonymous namespace
- [x] If parallelization is used, proper use of the abstracted complex classes are used. Using TBB specifically in a filter should be frowned upon unless for a really good reason.

## Filter Checklist

- [x] Parameters should be generally broken down into "Input Parameters", "Required Data Objects", "Created Data Objects". There can be exceptions to this.
- [x] ChoicesParameter selections should be an enumeration defined in the filer header
- [x] Documentation copied from SIMPL Repo and updated (if necessary)
- [x] Parameter argument variables are k_CamelCase_Key
- [x] Parameter argument strings are lower_snake_case
```
static inline constexpr StringLiteral k_AlignmentType_Key = "alignment_type";
```

## Unit Testing
- [x] 1 Unit test to instantiate the filter with the default arguments.
- [x] 1 Unit test to test output from the filter against known exemplar set of data
- [x] 1 Unit test to test invalid input code paths that are specific to a filter. Don't test that a DataPath does not exist since that test is already performed as part of the SelectDataArrayAction.

## Code Cleanup
- [x] No commented out code (rare exceptions to this is allowed..)
- [x] Filters should have both the Filter class and Algorithm class for anything beyond trivial needs
- [x] No API changes were made (or the changes have been approved)
- [x] No major design changes were made (or the changes have been approved)
- [x] Added test (or behavior not changed)
- [x] Updated API documentation (or API not changed)
- [x] Added license to new files (if any)
- [x] Added Python wrapping to new files (if any) as necessary
- [x] Added example pipelines that use the filter
- [x] Classes and methods are properly documented.

### CODE REVIEW

File: OrientationAnalysis/src/OrientationAnalysis/Filters/Algorithms/FindTriangleGeomShapes.cpp [hunk 1]
```diff
@@ -0,0 +1,458 @@
...
+                                   0.5 * (vertPtr[3 * vertIds[0] + 1] + vertPtr[3 * vertIds[1] + 1]),
+                                   0.5 * (vertPtr[3 * vertIds[0] + 2] + vertPtr[3 * vertIds[1] + 2]),
+                                   0.5 * (vertPtr[3 * vertIds[1] + 0] + vertPtr[3 * vertIds[2] + 0]),
+                                   0.5 * (vertPtr[3 * vertIds[1] + 1] + vertPtr[3 * vertIds[2] + 1]),
+                                   0.5 * (vertPtr[3 * vertIds[1] + 2] + vertPtr[3 * vertIds[2] + 2]),
+                                   0.5 * (vertPtr[3 * vertIds[2] + 0] + vertPtr[3 * vertIds[0] + 0]),
+                                   0.5 * (vertPtr[3 * vertIds[2] + 1] + vertPtr[3 * vertIds[0] + 1]),
+                                   0.5 * (vertPtr[3 * vertIds[2] + 2] + vertPtr[3 * vertIds[0] + 2])};
+  // clang-format off
+  std::array<int32_t, 32> tets = {
```
Comment (joeykleingers) [inline OrientationAnalysis/src/OrientationAnalysis/Filters/Algorithms/FindTriangleGeomShapes.cpp]:
```suggestion
  std::array<int32, 32> tets = {
```

File: OrientationAnalysis/src/OrientationAnalysis/Filters/Algorithms/FindTriangleGeomShapes.cpp [hunk 2]
```diff
@@ -0,0 +1,458 @@
...
+       1, 8, 7, 5,
+       2, 9, 8, 6,
+       7, 5, 6, 4,
+       6, 9, 7, 4,
+       6, 5, 7, 8,
+       7, 9, 6, 8
+      };
+  // clang-format on
+
+  for(size_t iter = 0; iter < 8; iter++)
```
Comment (joeykleingers) [inline OrientationAnalysis/src/OrientationAnalysis/Filters/Algorithms/FindTriangleGeomShapes.cpp]:
```suggestion
  for(usize iter = 0; iter < 8; iter++)
```

File: OrientationAnalysis/src/OrientationAnalysis/Filters/Algorithms/FindTriangleGeomShapes.cpp [hunk 3]
```diff
@@ -0,0 +1,458 @@
...
+constexpr usize k_12 = 5;
+constexpr usize k_20 = 6;
+constexpr usize k_21 = 7;
+constexpr usize k_22 = 8;
+
+// -----------------------------------------------------------------------------
+template <typename T>
+void FindTetrahedronInfo(const std::array<usize, 3>& vertIds, const DataArray<T>& vertPtr, const std::array<T, 3>& centroid, std::array<T, 32>& tetInfo)
+{
+  std::array<double, 30> coords = {vertPtr[3 * vertIds[0] + 0],
```
Comment (joeykleingers) [inline OrientationAnalysis/src/OrientationAnalysis/Filters/Algorithms/FindTriangleGeomShapes.cpp]:
```suggestion
  std::array<float64, 30> coords = {vertPtr[3 * vertIds[0] + 0],
```

File: OrientationAnalysis/src/OrientationAnalysis/Filters/Algorithms/FindTriangleGeomShapes.cpp [hunk 4]
```diff
@@ -0,0 +1,458 @@
...
+  const Float32Array& centroids = m_DataStructure.getDataRefAs<Float32Array>(m_InputValues->CentroidsArrayPath);
+  const Float32Array& volumes = m_DataStructure.getDataRefAs<Float32Array>(m_InputValues->VolumesArrayPath);
+
+  auto& omega3S = m_DataStructure.getDataRefAs<Float32Array>(m_InputValues->Omega3sArrayPath);
+
+  size_t numFaces = faceLabels.getNumberOfTuples();
+  size_t numFeatures = centroids.getNumberOfTuples();
+  m_FeatureMoments.resize(numFeatures * 6, 0.0);
+
+  std::array<float, 3> centroid = {0.0F, 0.0F, 0.0F};
```
Comment (joeykleingers) [inline OrientationAnalysis/src/OrientationAnalysis/Filters/Algorithms/FindTriangleGeomShapes.cpp]:
```suggestion
  std::array<float32, 3> centroid = {0.0F, 0.0F, 0.0F};
```

File: OrientationAnalysis/src/OrientationAnalysis/Filters/Algorithms/FindTriangleGeomShapes.cpp [hunk 5]
```diff
@@ -0,0 +1,458 @@
...
+  std::array<float, 32> tetInfo = {0.0F, 0.0F, 0.0F, 0.0F, 0.0F, 0.0F, 0.0F, 0.0F, 0.0F, 0.0F, 0.0F, 0.0F, 0.0F, 0.0F, 0.0F, 0.0F,
```
Comment (joeykleingers) [inline OrientationAnalysis/src/OrientationAnalysis/Filters/Algorithms/FindTriangleGeomShapes.cpp]:
```suggestion
  std::array<float32, 32> tetInfo = {0.0F, 0.0F, 0.0F, 0.0F, 0.0F, 0.0F, 0.0F, 0.0F, 0.0F, 0.0F, 0.0F, 0.0F, 0.0F, 0.0F, 0.0F, 0.0F,
```

File: OrientationAnalysis/src/OrientationAnalysis/Filters/Algorithms/FindTriangleGeomShapes.cpp [hunk 6]
```diff
@@ -0,0 +1,458 @@
...
+                                   0.0F, 0.0F, 0.0F, 0.0F, 0.0F, 0.0F, 0.0F, 0.0F, 0.0F, 0.0F, 0.0F, 0.0F, 0.0F, 0.0F, 0.0F, 0.0F};
+  std::array<usize, 3> vertIds = {0, 0, 0};
+
+  for(size_t i = 0; i < numFaces; i++)
```
Comment (joeykleingers) [inline OrientationAnalysis/src/OrientationAnalysis/Filters/Algorithms/FindTriangleGeomShapes.cpp]:
```suggestion
  for(usize i = 0; i < numFaces; i++)
```

**Review by joeykleingers — CHANGES_REQUESTED** [review]
Double check that every filter is using the proper complex type names.  I highlighted a few of them, but there are quite a few more spots that should be changed.

int8_t -> int8
uint8_t -> uint8
...
float -> float32
double -> float64
size_t -> usize

Other than that, this all looks good to me.

**Comment by imikejackson** [comment]
@imikejackson Test comment

**Comment by imikejackson** [comment]
@imikejackson Test Comment

**Comment by imikejackson** [comment]
@imikejackson Test commnet

**Comment by joeykleingers** [comment]
@imikejackson Test comment.

File: OrientationAnalysis/src/OrientationAnalysis/Filters/Algorithms/FindTriangleGeomShapes.cpp (hunk 1 above)
Comment (imikejackson) [reply OrientationAnalysis/src/OrientationAnalysis/Filters/Algorithms/FindTriangleGeomShapes.cpp]: @joeykleingers I disagree with our types.

**Review by joeykleingers — APPROVED** [review]
This PR looks good, and can be merged in once it passes all the usual checks.

**Pull request merged**